### PR TITLE
develop: Various chgres_cube bug fixes and code improvements

### PR DESCRIPTION
**Pull request by LarissaReames-NOAA**
This PR addresses all of the items listed in Issue #228 . Each item in that issue is addressed with its own commit.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@LarissaReames-NOAA The authoritative repo was updated this morning.  Merge these changes to your fork.  Then, I will do my final review and merge.

**Comment by LarissaReames-NOAA** [comment]
> @LarissaReames-NOAA The authoritative repo was updated this morning. Merge these changes to your fork. Then, I will do my final review and merge.

Done.

**Comment by GeorgeGayno-NOAA** [comment]
I compiled the branch at 840d5a8 on WCOSS-Dell.  All chgres_cube regression tests passed.

**Comment by LarissaReames-NOAA** [comment]
> I compiled the branch at [840d5a8](https://github.com/NOAA-EMC/UFS_UTILS/commit/840d5a83d6709565c1fb4d4bbab3416928fa01be) on WCOSS-Dell. All chgres_cube regression tests passed.

Anything left to change before merge?

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
Did you see my comments about the call to the error handler?  Also, 'develop' was updated this morning.  You may need to do a merge

File: sorc/chgres_cube.fd/input_data.F90 [hunk 1]
```diff
@@ -4875,40 +4875,42 @@ subroutine read_input_sfc_grib2_file(localpet)
...
-       ! failed => rc < 0
-       if (rc<0) print*, "ERROR FINDING SOILCTOP IN GEOGRID FILE"
-       if (rc == 0) then
-         print*, "READ SOIL TYPE FRACTIONS FROM GEOGRID FILE "
-         rc = nf90_get_var(ncid2d,varid,dummy3d_stype)
-         ! failed => rc < 0
-         if (rc<0) print*, "ERROR READING SCT_DOM FROM FILE"
-         print*, "min max dummy3d_stype = ", minval(dummy3d_stype), maxval(dummy3d_stype)
-       endif
+     if (rc /= nf90_noerr) call error_handler("READING GEOGRID FILE",rc)
```
[the remark this answers is not in the document]
Comment (LarissaReames-NOAA) [reply sorc/chgres_cube.fd/input_data.F90]: No, I was never notified of this comment, but it's been addressed in the latest commit.

**Comment by GeorgeGayno-NOAA** [comment]
Compiled the branch at 90b1f0d on WCOSS-Dell.  All chgres_cube regression tests passed.  Comments on error handling were addressed.  Will merge.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**